components/gateway/internal/gqlschema/schema.graphql

```diff
@@ -52,10 +54,60 @@ type Application {
     webhooks: [ApplicationWebhook!]!
     healthCheckURL: String
     """ group allows to find different versions of the same API """
-    apis(group: String): [APIDefinition!]!
+    apis(group: String, first: Int = 100, after: PageCursor): APIDefinitionPage!
     """ group allows to find different versions of the same event API """
-    eventAPIs(group: String): [EventAPIDefinition!]!
-    documents: [Document!]!
+    eventAPIs(group: String, first: Int = 100, after: PageCursor): EventAPIDefinitionPage!
+    documents(first: Int = 100, after: PageCursor): DocumentPage!
+}
+
+""" Every query that implements pagination returns object that implements Pageable interface.
+To specify page details, query specify two parameters: `first` and `after`.
+`first` specify page size, `after` is a cursor for the next page. When requesting first page, set `after` to empty value.
+For requesting next page, set `after` to `pageInfo.endCursor` returned from previous query. """
+interface Pageable {
+    pageInfo: PageInfo!
+    totalCount: Int!
```

Comment:
Do we really need `totalCount` property? Relay-style pagination doesn't have one. GraphQL client also doesn't need it - `hasNextPage` is sufficient.

Reply:
having `totalCount` is safer IMO. I can imagine that there will be requirements in the UI to display a number of pages.